monkestation/code/modules/hydroponics/machines/genesischamber.dm

```diff
@@ -0,0 +1,288 @@
+			return
+		if(!user.transferItemToLoc(W, src))
+			return
+		has_core = TRUE
+		to_chat(user, span_notice("You insert the vortex anomaly core into [src]."))
+		playsound(src, 'sound/machines/click.ogg', 50, TRUE)
+		return
+	return ..()
+
+/obj/machinery/genesis_chamber/attack_hand_secondary(mob/living/user, list/modifiers)
+	. = ..()
+	if(. == SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN)
+		return
+
+	if(obj_flags & EMAGGED)
+		return
+
+	if(panel_open)
+		return
+
+	if(!has_core)
+		to_chat(user, span_warning("The [src] needs a vortex anomaly core to function!"))
+		return
+
+	on = !on
+
+	if(on)
+		to_chat(user, span_notice("You turn on [src]."))
+		playsound(loc, 'sound/machines/chime.ogg', 40, TRUE)
+		say("Activating... Seeds will be produced in [DisplayTimeText(max_cooldown)].")
+		S_TIMER_COOLDOWN_START(src, "seed_generation", max_cooldown)
+		START_PROCESSING(SSprocessing, src)
+	else
+		to_chat(user, span_notice("You turn off [src]."))
+		playsound(loc, 'sound/machines/click.ogg', 15, TRUE, -3)
+		say("Deactivating...")
+		STOP_PROCESSING(SSprocessing, src)
+		if(TIMER_COOLDOWN_CHECK(src, "seed_generation"))
+			TIMER_COOLDOWN_END(src, "seed_generation")
+		time_remaining = max_cooldown
+
+	update_appearance()
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/machinery/genesis_chamber/process()
+	if(!on || !has_core)
+		return PROCESS_KILL
+
+	if(capacity >= max_capacity)
+		on = FALSE
+		say("Maximum capacity reached. Shutting down.")
+		playsound(src, 'sound/machines/ping.ogg', 50, TRUE)
+		update_appearance()
+		return PROCESS_KILL
+
+	if(!TIMER_COOLDOWN_CHECK(src, "seed_generation"))
+		generate_seed()
+		S_TIMER_COOLDOWN_START(src, "seed_generation", max_cooldown)
+
+	return TRUE
+
+/obj/machinery/genesis_chamber/proc/generate_seed()
+	if(!on || !has_core || capacity >= max_capacity)
+		return
+
+	// Calculate how many seeds to produce this cycle
+	var/seeds_to_produce = min(seeds_per_cycle, max_capacity - capacity)
+
+	if(seeds_to_produce <= 0)
+		return
+
+	capacity += seeds_to_produce
+
+	if(seeds_to_produce == 1)
+		say("Seed generated. Current capacity: [capacity]/[max_capacity]")
+	else
+		say("[seeds_to_produce] seeds generated. Current capacity: [capacity]/[max_capacity]")
+
+	playsound(src, 'sound/machines/synth_yes.ogg', 30, TRUE)
+	update_appearance()
+
+/obj/machinery/genesis_chamber/examine(mob/user)
+	. = ..()
+	if(!has_core)
+		. += "It is missing a <b>vortex anomaly core</b>."
+	else if(!on)
+		. += "It is currently <b>Deactivated</b> and has <b>[capacity]</b> strange seeds stored."
+	else
+		. += "It is currently <b>Activated</b> and has <b>[capacity]</b> strange seeds stored."
+		calculate_time_remaining()
+		. += "It'll take <b>[DisplayTimeText(time_remaining)]</b> until more seeds are created."
+	. += "The [src] can currently hold <b>[max_capacity] seeds</b> and produces <b>[seeds_per_cycle] seed[seeds_per_cycle == 1 ? "" : "s"]</b> every <b>[DisplayTimeText(max_cooldown)]</b>."
```

Comment:
```suggestion
	. += "It can currently hold <b>[max_capacity] seeds</b> and produces <b>[seeds_per_cycle] seed\s</b> every <b>[DisplayTimeText(max_cooldown)]</b>."
```